### ISSUE TITLE
Features request 

### ISSUE THREAD

**Issue by john-roland** [exceptionless/Exceptionless#276]
Hello!

Exceptionless is awesome, but here are some features we still miss (sorry for the lack of details) :

# Search #
* Have an advanced search screen, with fields to select tags, users, etc.
* Have the url been modified on search results, so we can share or bookmark it
* nice to have: keep a search history
* nice to have: save a search to reuse it later
* nice to have: add a saved filter as a new left menu entry (Filters / my filter 1…)
 
# Dashboard #
* have the possibility to add an new entry in the left side menu for our own event type, or that Exceptionless automatically detect it. in my case it's a "metric" type. 
* on stack dashboard, display a pie chart of tags distribution on the events of the stack. 
* on project & stack dashboard, display tags as filtering link, so a click on it will filters the events on the actual tag. 

# Events #
* Add a new tag system, with name/value pair. (ex: tag name: office, value: gva, ldn; tag name: matrix, tagname : database ; etc. etc;) and provide pie chart chart of tag value repartition on events 
* Better display of the events of a session (with the timestamp)
* List of unique users of a project

# Administration #

* in the project list, the numbers of events should use the current period filtering, now it’s the total.
* New chart: #events / weeks (and month) by project and by organisation
* Be able to move a project to another organization.. or maybe just be able to indicate my own "apikey" on a new project.
* Be able to export all the events of a project, on a time range, maybe by event  type. (maybe in the webAPI ?)
* Add a retention policy on each project, to define how long the events of a certain event type (log, exception feature usage..) should be kept
* Reset project data older than XXXX (we are running version < 4, and we don’t see any physical hard drive space recovery when we delete project data)
* User privilege (roles:  reader, developer, admin). Some people should not be able to delete data or project, at least :). to start a reader access would be great.
* nice to have: add the notion of sub projects

# Webapi #
* Improve the documentation (we’ve had to get the code to know all the parameters we can pass to submit event, for example)
* As said above, add a method to dump/export project events with details, on a time range, event type.

**Comment by niemyjski** [comment]
Thank you so much for posting this, this is great and helps us :).  From looking over this list many of these are things we already want do do with search and and dashboards. Would you like me to break down the things you can do currently in a a section or would you like to meet up and go over these (greatly appreciated). I do have some questions on some of these items.

**Comment by john-roland** [comment]
hello,
yes, we can meet up. This week has been too busy, maybe next one !

**Comment by niemyjski** [comment]
Yes, just let me know when works for you or send me a message on slack (preferred)

**Comment by niemyjski** [comment]
I agree, we need that. I need to really break all this down into separate issues and get your feedback @john-roland . If you just let me know a time when you are free I'll be there (as long as it's not 3am my time) :)

**Comment by john-roland** [comment]
Thursday, 5pm geneva time, should be 10am for you, would it be ok ?

**Comment by john-roland** [comment]
ho no sorry, already in a meeting this day... with our houston team ;)
this week it will be complicated.. or friday

**Comment by niemyjski** [comment]
Sounds good :), just send me an email (http://github.com/niemyjski) with the info or on slack :).

**Comment by niemyjski** [comment]
Thanks for having the team meet up with me, I greatly appreciate it. I need to split this into several new issues but here are my notes, please feel free to open new issues with any additional feedback. 

# Search
    - auto complete, would really like to be able to just bookmark search.
    - Anything else you want to search on?
    - Lists don't have project or date. (maybe make the tags on a separete line...)
    - Named tags? key value pair E.g., `Database:Europe`

# Dashboard
  - Type Dropdown in top search bar...
  - page size isn't configurable.. would like to see more on most recent on all events..
  - dashboards would be better with charts and not so list... timestamp... list of sessions is great, but not a unique list of users. (maybe group users with same name.)
  - sessions by users.
  - events by user.
  - how many users are using the app?
  - info added to tags via key value pair.. charts on dashboard for tag..
  - distribution over one tag... graph over different locations..

  - put more data in user metadata or environment and be able to search it and see it..
  - main point is cleaning data older than 3 months  keep feature usage.
  - just get the last three months of data. auto remove anything remove anything not cricial or discard by event type retention (just reduce noise).

  - daily summary is noisey (wants organization level daily summary)
  - clean all exceptions to get a new start.... and get new notifications again...
  - uses mark as critical as a way to reduce noise

**Comment by john-roland** [comment]
yes, definitively, if we are able to add some custom environment data, and be able to filter on it, it will be awesome.

I can imagine, in our usage, to add the following
- "Office" property
- "Company" property
- "Database" property
- "Matrix" (or team) property

etc.

maybe that already possible via a plugin, or not ?

etc..

**Comment by niemyjski** [comment]
You can do this (search by custom property) today if you call SetProperty() on the event builder with simple types (bool, string, int)

**Comment by ejsmith** [comment]
We're closing this as part of an open issue cleanup. This is an older grab-bag request, and the app has changed substantially since it was filed.

Any specific behavior that is still missing in the latest app or Svelte UI should be opened as a separate focused issue so it can be evaluated and implemented independently.
